code/FormSchemaController.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+namespace SilverStripe\Admin;
+
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Core\Validation\ValidationResult;
+use SilverStripe\Forms\Form;
+use SilverStripe\Forms\HTMLEditor\HTMLEditorConfig;
+use SilverStripe\Forms\HTMLEditor\TinyMCEConfig;
+use SilverStripe\Forms\Schema\FormSchema;
+use SilverStripe\Security\Security;
+
+/**
+ * The base class for controllers in the CMS that need access to form schema functionality.
+ */
+abstract class FormSchemaController extends AdminController
+{
+    /**
+     * Form schema header identifier
+     */
+    public const SCHEMA_HEADER = 'X-Formschema-Request';
+
+    private static array $allowed_actions = [
+        'schema',
+    ];
+
+    private static array $url_handlers = [
+        'GET schema/$FormName/$ItemID/$OtherItemID' => 'schema',
+    ];
+
+    private static array $dependencies = [
+        'FormSchema' => '%$' . FormSchema::class,
+    ];
+
+    /**
+     * Current form schema helper
+     */
+    private ?FormSchema $schema = null;
+
+    protected function init()
+    {
+        $this->beforeExtending('onInit', function () {
+            // Set the members html editor config
+            if (Security::getCurrentUser()) {
+                HTMLEditorConfig::set_active_identifier(Security::getCurrentUser()->getHtmlEditorConfigForCMS());
+            }
+
+            // Set default values in the config if missing.  These things can't be defined in the config
+            // file because insufficient information exists when that is being processed
+            $htmlEditorConfig = HTMLEditorConfig::get_active();
+            $htmlEditorConfig->setOption('language', TinyMCEConfig::get_tinymce_lang());
+            $langUrl = TinyMCEConfig::get_tinymce_lang_url();
+            if ($langUrl) {
+                $htmlEditorConfig->setOption('language_url', $langUrl);
+            }
+        });
+        parent::init();
+    }
```

Comment:
Needed for controllers such as `ElementalAreaController` which may include WYSIWYG in their forms.